docs/examples/notebooks/anim_utils.py

```diff
@@ -0,0 +1,180 @@
+    fig = plt.figure(figsize=figsize)
+    ax = plt.subplot(111)
+    ax.set_aspect('equal')
+
+    # Set the axis limits
+    axis_limits = max(max(np.absolute(x)), max(np.absolute(y))) * 1.2
+    ax.set_xlim(-axis_limits, axis_limits)
+    ax.set_ylim(-axis_limits, axis_limits)
+    ax.set_xlabel('')
+
+    # Make the initial line collection object
+    lsegs = LineCollection(make_2d_segments([0], [0]), cmap=cmap, lw=lw)
+    lsegs.set_array(np.arange(trail_length))
+    ax.add_collection(lsegs)
+
+    # Time label
+    time_str = ax.text(-axis_limits / 1.1, -axis_limits / 1.1, '',
+                       fontsize=12)
+
+    # Number of data points to skip per frame
+    skip = int(ms_per_frame / (1000.0 * dt))
+
+    def anim_init():
+        # Animation initialization function
+        lsegs.set_segments(None)
+        time_str.set_text('')
+        return lsegs, time_str
+
+    def animate_frame(i):
+        # Animiation function for each frame
+        lsegs.set_segments(
+            make_2d_segments(
+                x[max(i * skip - trail_length, 0):i * skip + 1],
+                y[max(i * skip - trail_length, 0):i * skip + 1]))
+        time_str.set_text('Time: %0.2fs' % (i * skip * dt))
+        return lsegs, time_str
+
+    num_frames = len(x) // skip
+    anim = animation.FuncAnimation(fig, animate_frame, init_func=anim_init,
+                                   frames=num_frames, interval=ms_per_frame,
+                                   blit=True)
+    plt.close(anim._fig)
+    return fig, ax, anim
+
+
+def make_anim_controlled_osc(x, y, w, dt=0.001, ms_per_frame=10,
+                             trail_length=500, figsize=(8, 6), cmap='gist_yarg',
+                             lw=2, dotsize=20):
+    def make_2d_segments(x_data, y_data):
+        # Convert the x, y arrays into line segments
+        vertices = np.array([x_data, y_data]).T.reshape(-1, 1, 2)
+        segments = np.concatenate([vertices[:-1], vertices[1:]], axis=1)
+        return segments
+
+    # Make matplotlib figure
+    fig = plt.figure(figsize=figsize)
+    ax = plt.subplot(111)
+    ax.set_aspect('equal')
+
+    # Set the axis limits
+    axis_limits = max(max(np.absolute(x)), max(np.absolute(y))) * 1.2
+    ax.set_xlim(-axis_limits, ((axis_limits / 1.2) + 0.5) * 1.2)
+    ax.set_ylim(-axis_limits, axis_limits)
+    ax.set_xticks(np.linspace(-int(axis_limits), int(axis_limits), 5))
+    ax.set_yticks(np.linspace(-int(axis_limits), int(axis_limits), 5))
+
+    # Make the initial line collection object
+    lsegs = LineCollection(make_2d_segments([0], [0]), cmap=cmap, lw=lw)
+    lsegs.set_array(np.arange(trail_length))
+    ax.add_collection(lsegs)
+
+    # Time label
+    time_str = ax.text(-axis_limits / 1.1, -axis_limits / 1.1, '',
+                       fontsize=12)
+
+    # Sidebar plot
+    axis_limits = max(np.absolute(w)) * 1.2
+    ax2 = ax.twinx()
+    ax2.set_ylim(-axis_limits, axis_limits)
+    dot, = ax2.plot([], [], 'r.', markersize=dotsize)

```

Comment:
Maybe we can put this in a subplot on the side instead of on the same plot. Maybe it's just me, but it looks kinda weird since the oscillator is off center?

Reply:
My initial plan was to have it in the subplot, but I couldn't get it to look right because of the 'equal' aspect setting on the oscillator plot (the two subplots wasn't the same height). Tried all sorts of things and nothing seemed to work. So I settled with this.

Reply:
I'll play around with it some more to see if I can find a better solution

Reply:
I think I've found a way to do it!